### ISSUE TITLE
Browser tests fail on Fedora 29

### ISSUE THREAD

**Issue by tsilvap** [EFForg/https-everywhere#17206]
Type: code issue
OS: Fedora 29 (GNU/Linux)

How to reproduce:
1. Clone `https-everywhere` repo locally and `cd` into the project root.
2. Run `bash install-dev-dependencies.sh` (no errors).
3. Run `bash test.sh`. It exits with this error message: `Chrome: HTTP to HTTPS redirection failed`

Here is the complete log if that's needed:
https://gist.github.com/tsilvap/e1feb74fa4730fa5c7f285e999d846ce

I have no idea what could be the issue here. I also tried this inside a pipenv running Python 3.6 (since Fedora uses Python 3.7 by default, I thought this could be the issue) but I get the same error.

**Comment by oleblaesing** [comment]
I'm also using Fedora 29 (Python 3.7) and I can't even run `bash install-dev-dependencies.sh` successfully, because I run into this error everytime I try. Please let me know if you need some more information about my environment.

```
https-everywhere [master] ✘ bash install-dev-dependencies.sh

Warning: Installing the development dependencies for HTTPS Everywhere
may alter your system, installing requirements both within the package
management system and also external binaries.

Are you sure you want to continue? [y/N]: y

Last metadata expiration check: 1:33:10 ago on So 03 Feb 2019 15:59:32 CET.
Package firefox-65.0-1.fc29.x86_64 is already installed.
Package gcc-8.2.1-6.fc29.x86_64 is already installed.
Package git-2.20.1-1.fc29.x86_64 is already installed.
Package libcurl-devel-7.61.1-7.fc29.x86_64 is already installed.
Package libxml2-devel-2.9.8-5.fc29.x86_64 is already installed.
Package libxslt-devel-1.1.32-3.fc29.x86_64 is already installed.
Package python2-devel-2.7.15-11.fc29.x86_64 is already installed.
Package redhat-rpm-config-118-1.fc29.noarch is already installed.
Package xorg-x11-server-Xvfb-1.20.3-3.fc29.x86_64 is already installed.
Package which-2.21-11.fc29.x86_64 is already installed.
Package findutils-1:4.6.0-21.fc29.x86_64 is already installed.
Package procps-ng-3.3.15-4.fc29.x86_64 is already installed.
Package openssl-1:1.1.1a-1.fc29.x86_64 is already installed.
Package openssl-devel-1:1.1.1a-1.fc29.x86_64 is already installed.
Package chromium-71.0.3578.98-1.fc29.x86_64 is already installed.
Package GConf2-3.2.6-21.fc29.x86_64 is already installed.
Dependencies resolved.
Nothing to do.
Complete!
Requirement already satisfied: lxml>=3.3.3 in /home/solitude/.local/lib/python3.7/site-packages (from -r requirements.txt (line 1)) (4.3.0)
Requirement already satisfied: lxml>=3.3.3 in /home/solitude/.local/lib/python3.7/site-packages (from -r requirements.txt (line 1)) (4.3.0)
Requirement already satisfied: pycurl in /usr/lib64/python3.7/site-packages (from -r requirements.txt (line 2)) (7.43.0.2)
Collecting regex (from -r requirements.txt (line 3))
  Using cached https://files.pythonhosted.org/packages/aa/eb/8a56aaf3a0a2a70cf2e017a8fb1ac5b6bad64a143d3096b0c0282b17ead1/regex-2019.01.24.tar.gz
Collecting bsdiff4 (from -r requirements.txt (line 4))
  Using cached https://files.pythonhosted.org/packages/12/a7/e76afa32f205469e87a0454e731bc9d57f04237da4c0e1d0801a912a1940/bsdiff4-1.1.5.tar.gz
Collecting python-Levenshtein (from -r requirements.txt (line 5))
  Using cached https://files.pythonhosted.org/packages/42/a9/d1785c85ebf9b7dfacd08938dd028209c34a0ea3b1bcdb895208bd40a67d/python-Levenshtein-0.12.0.tar.gz
Collecting tldextract (from -r requirements.txt (line 6))
  Using cached https://files.pythonhosted.org/packages/0e/0c/1b7332684dfc2e6311d59cd00859a5318a7e0ba50334ad217ceb9555e213/tldextract-2.2.0-py2.py3-none-any.whl
Requirement already satisfied: setuptools in /usr/lib/python3.7/site-packages (from python-Levenshtein->-r requirements.txt (line 5)) (40.4.3)
Requirement already satisfied: idna in /usr/lib/python3.7/site-packages (from tldextract->-r requirements.txt (line 6)) (2.7)
Requirement already satisfied: requests>=2.1.0 in /usr/lib/python3.7/site-packages (from tldextract->-r requirements.txt (line 6)) (2.20.0)
Requirement already satisfied: requests-file>=1.4 in /usr/lib/python3.7/site-packages (from tldextract->-r requirements.txt (line 6)) (1.4.3)
Requirement already satisfied: chardet<3.1.0,>=3.0.2 in /usr/lib/python3.7/site-packages (from requests>=2.1.0->tldextract->-r requirements.txt (line 6)) (3.0.4)
Requirement already satisfied: urllib3<1.25,>=1.21.1 in /usr/lib/python3.7/site-packages (from requests>=2.1.0->tldextract->-r requirements.txt (line 6)) (1.24.1)
Requirement already satisfied: six in /usr/lib/python3.7/site-packages (from requests-file>=1.4->tldextract->-r requirements.txt (line 6)) (1.11.0)
Installing collected packages: regex, bsdiff4, python-Levenshtein, tldextract
  Running setup.py install for regex ... error
    Complete output from command /usr/bin/python3 -u -c "import setuptools, tokenize;__file__='/tmp/pip-install-ua7hdzew/regex/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /tmp/pip-record-25w3_z0_/install-record.txt --single-version-externally-managed --compile --user --prefix=:
    /usr/lib/python3.7/site-packages/setuptools/dist.py:398: UserWarning: Normalizing '2019.01.24' to '2019.1.24'
      normalized_version,
    running install
    running build
    running build_py
    creating build
    creating build/lib.linux-x86_64-3.7
    copying regex_3/regex.py -> build/lib.linux-x86_64-3.7
    copying regex_3/_regex_core.py -> build/lib.linux-x86_64-3.7
    copying regex_3/test_regex.py -> build/lib.linux-x86_64-3.7
    running build_ext
    building '_regex' extension
    creating build/temp.linux-x86_64-3.7
    creating build/temp.linux-x86_64-3.7/regex_3
    gcc -pthread -Wno-unused-result -Wsign-compare -DDYNAMIC_ANNOTATIONS_ENABLED=1 -DNDEBUG -O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions -fstack-protector-strong -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -D_GNU_SOURCE -fPIC -fwrapv -fPIC -I/usr/include/python3.7m -c regex_3/_regex.c -o build/temp.linux-x86_64-3.7/regex_3/_regex.o
    regex_3/_regex.c:46:10: fatal error: Python.h: No such file or directory
     #include "Python.h"
              ^~~~~~~~~~
    compilation terminated.
    error: command 'gcc' failed with exit status 1

    ----------------------------------------
Command "/usr/bin/python3 -u -c "import setuptools, tokenize;__file__='/tmp/pip-install-ua7hdzew/regex/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /tmp/pip-record-25w3_z0_/install-record.txt --single-version-externally-managed--compile --user --prefix=" failed with error code 1 in /tmp/pip-install-ua7hdzew/regex/
```

**Comment by Bisaloo** [comment]
@oleblaesing could you install `python3-devel` and see if it fixes the issue please?

**Comment by oleblaesing** [comment]
Yeah, that fixes the problem (thank you). 👍

**Comment by J0WI** [comment]
I think this has been fixed in #17494
